Bottom: eb35bccb6ed76896fb86cd18bdc7ef9d7c235193
Top:    296f1c2066c8f8c7729ae3381286d6e689b1d40c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-30 00:04:34 +0100

draw.c: Have DrawLayerGroup() call DrawPPV for non-gui exporters

This means DrawLayerGroup() can have a void return type as most of
the other drawing functions, and makes things a little neater.

Only the GUI renderers special case the pin / pad / via to wait
until a later rendering oass than the layers they reside on.


---

diff --git a/src/draw.c b/src/draw.c
index 1bbbca0..83080b3 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -83,7 +83,7 @@ static bool doing_assy = false;
  */
 static void DrawEverything (BoxTypePtr);
 static void DrawPPV (int group, const BoxType *);
-static int DrawLayerGroup (int, const BoxType *);
+static void DrawLayerGroup (int, const BoxType *);
 static void AddPart (void *);
 static void SetPVColor (PinTypePtr, int);
 static void DrawEMark (ElementTypePtr, Coord, Coord, bool);
@@ -612,7 +612,6 @@ PrintAssembly (int side, const BoxType * drawn_area)
   doing_assy = true;
   gui->set_draw_faded (Output.fgGC, 1);
   DrawLayerGroup (side_group, drawn_area);
-  DrawPPV (side_group, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
@@ -677,8 +676,7 @@ DrawEverything (BoxTypePtr drawn_area)
 
       if (gui->set_layer (0, group, 0))
         {
-          if (DrawLayerGroup (group, drawn_area) && !gui->gui)
-            DrawPPV (group, drawn_area);
+          DrawLayerGroup (group, drawn_area);
           gui->end_layer ();
         }
     }
@@ -1119,10 +1117,10 @@ DrawLayer (LayerTypePtr Layer, const BoxType * screen)
 }
 
 /* ---------------------------------------------------------------------------
- * draws one layer group.  Returns non-zero if pins and pads should be
- * drawn with this group.
+ * draws one layer group.  If the exporter is not a GUI,
+ * also draws the pins / pads / vias in this layer group.
  */
-static int
+static void
 DrawLayerGroup (int group, const BoxType *drawn_area)
 {
   int i, rv = 1;
@@ -1143,7 +1141,9 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
     }
   if (n_entries > 1)
     rv = 1;
-  return rv;
+
+  if (rv && !gui->gui)
+    DrawPPV (group, drawn_area);
 }
 
 static void
